src/main/java/org/yorm/util/RowRecordConverter.java

```diff
@@ -1,90 +1,343 @@
+                    throw new IllegalStateException(
+                            String.format(
+                                    "Unexpected exception invoking %s.value()[%d].",
+                                    outputTypeClass.getName(),
+                                    (int) input
+                            ),
+                            e
+                    );
+                }
+            };
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            // Again... This should never happen, but let's give helpful output just in case weird stuff goes down!
+            throw new IllegalStateException(
+                    String.format("Unexpected exception finding %s.value()[] method.", outputTypeClass.getName()),
+                    e
+            );
+        }
+    }
+
+    // TODO: Cache for each enum type? I could imagine times when we'd be calling this a lot for the same thing.
+    private static <OutputType> Converter<OutputType> enumFromString(Class<OutputType> outputTypeClass) {
+        try {
+            MethodHandle methodHandle = MethodHandles
+                    .lookup()
+                    .findStatic(
+                            outputTypeClass,
+                            "valueOf",
+                            MethodType.methodType(outputTypeClass, String.class)
+                    );
+            return input -> {
+                try {
+                    // noinspection unchecked
+                    return (OutputType) methodHandle.invoke((String) input);
+                } catch (IllegalArgumentException e) {
+                    // Rethrow real errors...
+                    throw e;
+                } catch (Throwable e) {
+                    // This should never happen, but let's give helpful output just in case weird stuff goes down!
+                    throw new IllegalStateException(
+                            String.format(
+                                    "Unexpected exception invoking %s.valueOf(\"%s\").",
+                                    outputTypeClass.getName(),
+                                    (String) input
+                            ),
+                            e
+                    );
+                }
+            };
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            // Again... This should never happen, but let's give helpful output just in case weird stuff goes down!
+            throw new IllegalStateException(
+                    String.format("Unexpected exception finding %s.valueOf() method.", outputTypeClass.getName()),
+                    e
+            );
+        }
+    }
+
+    public void recordToRow(
+            int paramIndex,
+            PreparedStatement preparedStatement,
+            String dbColumnName,
+            Object value,
+            DbType type,
+            Converter<?> converter
+    ) throws SQLException, YormException {
```

Comment:
It seems the YormException is not used in this case, could be removed

Reply:
I just removed the catch! :)